backend/code_groups/cg_controllers.js

```diff
@@ -254,15 +255,31 @@ exports.updateGroupDetails = async (req, res) => {
 
             const userSocketId = userSocketMap.get(potentialMember.id);
             if (userSocketId) {
-              userSocketId.emit("added_user_to_group", {
+              try {
+                await rateLimiter.consume(potentialMember.id);
+
+                userSocketId.emit("added_user_to_group", {
+                  message: `You have been added to the code group: ${group.groupName}`,
+                });
+              } catch (rejRes) {
+                socket.emit("Too many notifications. Please try again later.");
+              }
+            }
+            await prisma.notification.create({
+              data: {
                 message: `You have been added to the code group: ${group.groupName}`,
-                userId: potentialMember.id,
-                senderId: userId,
-                category: "added_to_code_group",
-                groupId: group.id,
+                category: notif_categories.added_to_group,
                 isImportant: true,
-              });
-            }
+                isOffline: !userSocketId && true,
```

Comment:
Why do you add the `&& true`? Checking the socketID alone should be sufficient

Reply:
@azhiah It assigns the value of 'true' to isOffline if userSocketId is not found. without "&& true", it would be an incomplete statement

Reply:
So with just `! userSocketId` it was giving you an error? Can you do something like `userSocketId == null` instead then? That's a clearer statement